File: app/modules/IArchiveRepository.java
package modules;

import models.IArchive;
import models.IDissemination;

import java.util.List;

/**
 * Created by Beemen on 06/08/2015.
 */
public interface IArchiveRepository {
    List<IDissemination> GetDIPs(IArchive archive);
    IDissemination LookupDIP(String keyString);
}

File: Modules/IArchiveRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using end_user_gui.Models;

namespace end_user_gui.Modules
{
    public interface IArchiveRepository
    {
        List<IDissemination> GetDIPs(Archive archive);
        IDissemination LookupDIP(String keyString);
        string FileUrl(string referenceCode, string path);
    }
}